### PR TITLE
Add perspective in log-det cone

### LINKED ISSUE #546: logdet cone definition is wrong (needs perspective variable)
the logdet function is not homogeneous, hence its hypograph (t,X) : t <= logdet(X) is convex but not conic. the "logdet cone" is therefore a misnomer. we should redefine the logdet cone as (x, y, Z) : x <= y*logdet(Z/y).

I'm happy to go through and do this.

### PR DESCRIPTION
**Pull request by blegat**
Rebased #553 and made the change suggested in https://github.com/JuliaOpt/MathOptInterface.jl/pull/553#discussion_r229964953

Closes #546 
Closes #553

### CODE REVIEW

File: src/Bridges/detbridge.jl [hunk 1]
```diff
@@ -7,59 +7,61 @@ function trimap(i::Integer, j::Integer)
...
-function extract_eigenvalues(model, f::MOI.VectorAffineFunction{T}, d::Int) where T
-    n = trimap(d, d)
-    N = trimap(2d, 2d)
-
-    Δ = MOI.add_variables(model, n)
-
+function extract_eigenvalues(model, f::MOI.VectorAffineFunction{T}, d::Int, offset::Int) where T
     f_scalars = MOIU.eachscalar(f)
-    X = f_scalars[2:(n+1)]
+    tu = [f_scalars[i] for i in 1:offset]
```
Comment (odow) [inline src/Bridges/detbridge.jl]: `tu = f_scalars[1:offset]`?

File: src/Bridges/detbridge.jl [hunk 2]
```diff
@@ -7,59 +7,61 @@ function trimap(i::Integer, j::Integer)
...
+
+    n = trimap(d, d)
+    X = f_scalars[offset+1:n+offset]
```
Comment (odow) [inline src/Bridges/detbridge.jl]: Style: spaces around binary operators

File: src/Bridges/detbridge.jl [hunk 3]
```diff
@@ -7,59 +7,61 @@ function trimap(i::Integer, j::Integer)
...
+
     cur = m
     for j in 1:d
         for i in j:d
             cur += 1
             terms[cur] = MOI.VectorAffineTerm(trimap(i, d+j), MOI.ScalarAffineTerm(one(T), Δ[trimap(i, j)]))
         end
         cur += 1
-            terms[cur] = MOI.VectorAffineTerm(trimap(d+j, d+j), MOI.ScalarAffineTerm(one(T), Δ[trimap(j, j)]))
+        terms[cur] = MOI.VectorAffineTerm(trimap(d+j, d+j), MOI.ScalarAffineTerm(one(T), Δ[trimap(j, j)]))
```
Comment (odow) [inline src/Bridges/detbridge.jl]: spaces again

**Comment by mlubin** [comment]
Anything else needed here? @chriscoey, does this look good to you?

**Comment by chriscoey** [comment]
lgtm! thanks for finishing it @blegat

**Pull request merged**